solr/core/src/test/org/apache/solr/security/JWTIssuerConfigTest.java

```diff
@@ -125,6 +128,31 @@ public void parseIssuerConfigExplicit() {
     assertEquals("https://host/jwk", issuerConfig.getJwksUrls().get(0));
   }
 
+  @Test
```

Comment:
If you want to stand up a lightweight server, you can do something like this. You'll probably want to randomize the port (maybe you can pass 0 to always get an open port?)

```
  @SuppressForbidden(reason="using httpserver to mock a response")
  @Test
  public void wellKnownConfigWithHttpLogsAWarning() throws IOException {
    com.sun.net.httpserver.HttpServer server = HttpServer.create(new InetSocketAddress("127.0.0.1", 45678), 1);
    server.createContext("/", exchange -> {  
      Path configJson = TEST_PATH().resolve("security").resolve("jwt_well-known-config.json"); 
      exchange.sendResponseHeaders(200, 0);
      exchange.getResponseBody().write(Files.readAllBytes(configJson));
      exchange.close();
    });
    try {
      server.start();

     // stuff
    } finally {
      server.stop(0);
    }
  }
```

Reply:
This feels like using a sledgehammer on a fly...   Thoughts on just coming up with a mock in the right place?   How mcuh change would it be to introduce a mock object to return the content instead of a full blown server.   Plus, this feels like it might be slow ;-)

Reply:
So another option would be to create a mock URL object, that returns a FileInputStream when called openConnection.

Reply:
Wait, that wouldn't work because we don't create the URL in the test, it's done in the main code. Well, maybe you could still thread this through, I'm not sure.